pkg/descheduler/setupplugins.go

```diff
@@ -31,20 +31,20 @@ import (
 )
 
 func SetupPlugins() {
-	pluginbuilder.PluginRegistry = pluginbuilder.NewRegistry()
-	RegisterDefaultPlugins(pluginbuilder.PluginRegistry)
+	pluginregistry.PluginRegistry = pluginregistry.NewRegistry()
+	RegisterDefaultPlugins(pluginregistry.PluginRegistry)
 }
 
-func RegisterDefaultPlugins(registry pluginbuilder.Registry) {
-	pluginbuilder.Register(defaultevictor.PluginName, defaultevictor.New, &defaultevictor.DefaultEvictorArgs{}, registry)
-	pluginbuilder.Register(nodeutilization.LowNodeUtilizationPluginName, nodeutilization.NewLowNodeUtilization, &nodeutilization.LowNodeUtilizationArgs{}, registry)
-	pluginbuilder.Register(nodeutilization.HighNodeUtilizationPluginName, nodeutilization.NewHighNodeUtilization, &nodeutilization.HighNodeUtilizationArgs{}, registry)
-	pluginbuilder.Register(podlifetime.PluginName, podlifetime.New, &podlifetime.PodLifeTimeArgs{}, registry)
-	pluginbuilder.Register(removeduplicates.PluginName, removeduplicates.New, &removeduplicates.RemoveDuplicatesArgs{}, registry)
-	pluginbuilder.Register(removefailedpods.PluginName, removefailedpods.New, &removefailedpods.RemoveFailedPodsArgs{}, registry)
-	pluginbuilder.Register(removepodshavingtoomanyrestarts.PluginName, removepodshavingtoomanyrestarts.New, &removepodshavingtoomanyrestarts.RemovePodsHavingTooManyRestartsArgs{}, registry)
-	pluginbuilder.Register(removepodsviolatinginterpodantiaffinity.PluginName, removepodsviolatinginterpodantiaffinity.New, &removepodsviolatinginterpodantiaffinity.RemovePodsViolatingInterPodAntiAffinityArgs{}, registry)
-	pluginbuilder.Register(removepodsviolatingnodeaffinity.PluginName, removepodsviolatingnodeaffinity.New, &removepodsviolatingnodeaffinity.RemovePodsViolatingNodeAffinityArgs{}, registry)
-	pluginbuilder.Register(removepodsviolatingnodetaints.PluginName, removepodsviolatingnodetaints.New, &removepodsviolatingnodetaints.RemovePodsViolatingNodeTaintsArgs{}, registry)
-	pluginbuilder.Register(removepodsviolatingtopologyspreadconstraint.PluginName, removepodsviolatingtopologyspreadconstraint.New, &removepodsviolatingtopologyspreadconstraint.RemovePodsViolatingTopologySpreadConstraintArgs{}, registry)
+func RegisterDefaultPlugins(registry pluginregistry.Registry) {
+	pluginregistry.Register(defaultevictor.PluginName, defaultevictor.New, &defaultevictor.DefaultEvictorArgs{}, defaultevictor.ValidateDefaultEvictorArgs, defaultevictor.SetDefaults_DefaultEvictorArgs, registry)
```

Comment:
This makes us go back to having switch cases in validateDeschedulerConfiguration and setDefaultsPluginConfig. Since we are dealing with runtime.Objects there is no way to invoke SetDefaults before checking the type of PluginArgInstance. Even if they are args methods, were are dealing with runtime.Objects, and you would not have access to its methods withough casting it first to the correct arg type. I personally prefer to keep this being registered just one time in the registry, and keeping  validateDeschedulerConfiguration and setDefaultsPluginConfig simple, without having to verify types in policyconfig.go